src/packages/core/efuns_main.cc

```diff
@@ -351,12 +351,25 @@ void f_capitalize(void) {
 #endif
 
 #ifdef F_CHILDREN
+//TODO:fix this. I basically moved code from the old children into this function but it should probably
+//be some sort of helper in array.h which works for all types. In future this "fix" probably won't be 
+//if I make changes to array_t.
 void f_children(void) {
-  array_t *vec;
+  auto max_array_size { CONFIG_INT(__MAX_ARRAY_SIZE__ ) };
+  auto v { ObjectTable::get()->children(sp->u.string) };
+  auto len { v.size() < max_array_size ? v.size() : max_array_size };
+  auto res {allocate_empty_array(len)};
+  auto i {0};
+
+  for(;i < v.size() && i < max_array_size;++i) {
+    res->item[i].u.ob = v[i];
+    res->item[i].type = T_OBJECT;
+    add_ref(v[i], "children");
+  }
+  res = resize_array(res, i);
```

Comment:
This has not been fixed, now you are allocating the correct size from begining, there are no need for resize array later.

Also, please don't code C89 style in c++ code, no need to separate declarations and statements.  

Specifically,  i can be changed into a scoped variable 

```for( int i =0; i< v.size() ; i++) {} ```